controllers/cleanup_controller.go

```diff
@@ -103,12 +103,34 @@ func (r *CleanupReconciler) isOcsOperatorSubAndCSVAt419(ctx context.Context) err
 	} else if strings.HasSuffix(ocsOperatorSubscription.Spec.Channel, "4.18") {
 		return fmt.Errorf("subscription of 'ocs-operator' still points to '4.18'")
 	} else if !strings.HasSuffix(ocsOperatorSubscription.Spec.Channel, "4.19") {
-		// a guard that this code should be skipped in 4.19 even if the code isn't removed
+		// a guard that this code should be skipped in 4.19+ even if the code isn't removed
 		return nil
 	}
 	if !strings.Contains(ocsOperatorSubscription.Status.InstalledCSV, "4.19") {
 		return fmt.Errorf("waiting for 'ocs-operator' installed CSV at '4.19'")
 	}
+
+	return r.isOcsCsvReady(ctx, logger, ocsOperatorSubscription.Status.InstalledCSV)
+}
+
+func (r *CleanupReconciler) isOcsCsvReady(ctx context.Context, logger logr.Logger, ocsCsvName string) error {
+
+	ocsCsv := &opv1a1.ClusterServiceVersion{}
+	ocsCsv.Name = ocsCsvName
+	ocsCsv.Namespace = OperatorNamespace
+
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(ocsCsv), ocsCsv)
+	if err != nil {
+		logger.Error(err, "failed getting ocs csv", "csvName", ocsCsv)
+		return err
+	}
+
+	if ocsCsv.Status.Phase != opv1a1.CSVPhaseSucceeded {
+		err = fmt.Errorf("csv %s is not in succeeded state", ocsCsvName)
+		logger.Error(err, "waiting for csv to be in succeeded state")
```

Comment:
no new error here, the error returned here anyways shows up in reconciler. so, no need to explicitly log in here.

Reply:
Reconciler errors are logged after the full reconcile call is completed. It is possible that a few more logs will be populated depending on the error handling, so it is better to log the error where it occurs for ease of debugging.